### PR TITLE
TMM-3: Add `--no-TTY` to `docker-compose exec` commands to fix validation scripts in newer versions on docker

### PR DESCRIPTION
**Pull request by rspurgeon**
### Description 

https://confluentinc.atlassian.net/browse/DEVX-2720

DEVX-2720: fixes for failed runs on modern docker and docker-compose
I have not been able to isolate why these changes are required on my development machine. However, without these the cp-demo start script failed for a variety of reasons.  The return values from executions of docker-compose exec ... resulted in empty variables which evaluated to various runtime error in the start script. 

This set of changes does not directly address the issues identified in DEVX-2720 but instead was required by me to get the demo to run at all.  I will next attempt to understand and address the issues noted in the Jira issue.'

### Author Validation
- [X] Documentation
- [X] Run cp-demo

### Reviewer Tasks
- [ ] Documentation
- [ ] Run cp-demo

### CODE REVIEW

**Review by ybyzek — DISMISSED** [review]
@rspurgeon thanks for the PR. I haven't tested it yet but to note another required change: the validation scripts should also be updated, where relevant.  For example: https://github.com/confluentinc/cp-demo/blob/7.1.x/scripts/validate/validate_rest_proxy.sh

**Comment by chuck-confluent** [comment]
I ran into the same and found `-T` option on docker compose exec did the trick https://github.com/confluentinc/cp-demo/issues/415

**Comment by chuck-confluent** [comment]
@rspurgeon is this ready to take out of draft and merge? If so, maybe I can shadow you on the merge process? I've not used `pint merge` before and it looks like the tools team now requires us to use the "pint merge service" rather than CLI. For context, TMM is taking ownership of CP Demo maintenance and development, so it would be instructive for me to look over your shoulder as we get this merged.

**Comment by chuck-confluent** [comment]
Lines to change to use the --no-TTY:
- https://github.com/confluentinc/cp-demo/blob/DEVX-2720/scripts/validate/validate_rest_proxy.sh#L49
- https://github.com/confluentinc/cp-demo/blob/DEVX-2720/scripts/validate/validate_rest_proxy.sh#L99
- https://github.com/confluentinc/cp-demo/blob/DEVX-2720/scripts/validate/validate_rest_proxy.sh#L101

**Review by chuck-confluent — CHANGES_REQUESTED** [review]
Lines to change to use the --no-TTY option with `docker-compose exec`:
- [x] https://github.com/confluentinc/cp-demo/blob/DEVX-2720/scripts/validate/validate_rest_proxy.sh#L49
- [x] https://github.com/confluentinc/cp-demo/blob/DEVX-2720/scripts/validate/validate_rest_proxy.sh#L99
- [x] https://github.com/confluentinc/cp-demo/blob/DEVX-2720/scripts/validate/validate_rest_proxy.sh#L101
- [x] https://github.com/confluentinc/cp-demo/blame/DEVX-2720/docs/hybrid-cloud.rst#L242-L243
- [x] https://github.com/confluentinc/cp-demo/blame/DEVX-2720/docs/on-prem.rst#L1206
- [ ] https://github.com/confluentinc/cp-demo/blame/DEVX-2720/docs/on-prem.rst#L1227
- [ ] https://github.com/confluentinc/cp-demo/blame/DEVX-2720/docs/on-prem.rst#L1435

**Comment by chuck-confluent** [comment]
By the way, gitpod prebuilds sort of act like CI, so I can actually see which branches hit this bug. The earliest branch where the prebuild timed out because of the bug was 6.2.0-post, so you should be able to use that as the target branch for this PR

image

**Review by chuck-confluent — APPROVED** [review]
approving

File: docs/on-prem.rst [hunk 1]
```diff
@@ -1432,7 +1432,7 @@ Before running this section:
 
    .. sourcecode:: bash
 
-      docker-compose exec kafka1 kafka-replica-status \
+      docker exec kafka1 kafka-replica-status \
```
Comment (ybyzek) [inline docs/on-prem.rst:1435]: @chuck-confluent it's post merge but why was this line changed?

Comment (chuck-confluent) [reply docs/on-prem.rst:1435]: I played with it a bit (probably even more than I should). The behavior of docker exec is different than docker-compose exec. We set the container name, so it's equivalent to docker-compose exec except the output pipes into grep more nicely.

**Comment by ybyzek** [comment]
@chuck-confluent I tested 7.1.0-post with this PR and it fails in my environment:

Change with PR: there is no output

```
$ docker-compose exec --no-TTY connect curl -X GET --cert /etc/kafka/secrets/connect.certificate.pem --key /etc/kafka/secrets/connect.key --tlsv1.2 --cacert /etc/kafka/secrets/snakeoil-ca-1.crt -u superUser:superUser https://connect:8083/connectors/$connectorName/status 2>/dev/null | jq -r .connector.state
```

Original code: output:

```
$ docker-compose exec connect curl -X GET --cert /etc/kafka/secrets/connect.certificate.pem --key /etc/kafka/secrets/connect.key --tlsv1.2 --cacert /etc/kafka/secrets/snakeoil-ca-1.crt -u superUser:superUser https://connect:8083/connectors/$connectorName/status 2>/dev/null | jq -r .connector.state 
RUNNING
```

tl;dr this PR breaks cp-demo and it should be reverted/fixed

**Comment by chuck-confluent** [comment]
Docker made a breaking change here, so it seems we have to choose to support the latest version of docker or older versions of docker. I will file a bug report with docker because this is a pretty shocking change in behavior. I will create a new issue to add a validation check for docker version. Thanks Yeva!
